diff --git a/sbin/patchman b/sbin/patchman
index ef69ad6..da96c95 100755
--- a/sbin/patchman
+++ b/sbin/patchman
@@ -311,8 +311,7 @@ def host_updates_alt(host=None):
                 phosts.append(fhost)
 
             for phost in phosts:
-                phost.updates.clear()
-                phost.updates = updates
+                phost.updates.set(updates)
                 phost.updated_at = ts
                 phost.save()
                 updated_hosts.append(phost)
